server/src/com/thoughtworks/go/server/controller/AgentRegistrationController.java

```diff
@@ -233,6 +233,10 @@ public ResponseEntity agentRequest(@RequestParam("hostname") String hostname,
                 return new ResponseEntity<>("Elastic agents must submit both elasticAgentId and elasticPluginId.", UNPROCESSABLE_ENTITY);
             }
 
+            if (elasticAgentIdAlreadyRegistered(elasticAgentId, elasticPluginId)) {
```

Comment:
Can we move this check inside [AgentConfig#validate](https://github.com/gocd/gocd/blob/master/config/config-api/src/com/thoughtworks/go/config/AgentConfig.java#L83-L111) ?